OSError: libippcore.so.10

### Summary

There is an incompatibility between `ipp 2022.0.0` and current `cil`. It is not yet reflected in the cil dependencies. We need a work around in MI until a new version of `cil` is released.

https://github.com/TomographicImaging/CIL/issues/1970

### Steps To Reproduce

Seen on github actions tests

### Expected Behaviour

Tests pass

### Current Behaviour

Error when importing cil

### Context

Current main

### Failure Logs

<!--- [Optional] Please include any relevant log snippets here. -->
```
 ConftestImportFailure: OSError: libippcore.so.10: cannot open shared object file: No such file or directory (from /home/runner/work/mantidimaging/mantidimaging/conftest.py)
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ImportError while loading conftest '/home/runner/work/mantidimaging/mantidimaging/conftest.py'.
conftest.py:11: in <module>
    from mantidimaging.core.utility.leak_tracker import leak_tracker
mantidimaging/core/__init__.py:5: in <module>
    from . import operations, io, parallel, utility  # noqa: F401
mantidimaging/core/io/__init__.py:5: in <module>
    from . import (  # noqa: F401
mantidimaging/core/io/saver.py:20: in <module>
    from ..operations.rescale import RescaleFilter
mantidimaging/core/operations/rescale/__init__.py:5: in <module>
    from .rescale import RescaleFilter  # noqa:F401
mantidimaging/core/operations/rescale/rescale.py:11: in <module>
    from mantidimaging.gui.utility.qt_helpers import Type
mantidimaging/gui/__init__.py:5: in <module>
    from .gui import execute  # noqa: F401  # noqa:F821
mantidimaging/gui/gui.py:13: in <module>
    from mantidimaging.gui.windows.main import MainWindowView
mantidimaging/gui/windows/main/__init__.py:6: in <module>
    from .view import MainWindowView  # noqa: F401
mantidimaging/gui/windows/main/view.py:38: in <module>
    from mantidimaging.gui.windows.recon import ReconstructWindowView
mantidimaging/gui/windows/recon/__init__.py:5: in <module>
    from .model import ReconstructWindowModel  # noqa: F401
mantidimaging/gui/windows/recon/model.py:12: in <module>
    from mantidimaging.core.reconstruct import get_reconstructor_for
mantidimaging/core/reconstruct/__init__.py:9: in <module>
    from .cil_recon import CILRecon
mantidimaging/core/reconstruct/cil_recon.py:13: in <module>
    from cil.framework import (AcquisitionData, AcquisitionGeometry, DataOrder, ImageGeometry, BlockGeometry,
../../../miniconda3/envs/mantidimaging-dev/lib/python3.12/site-packages/cil/framework/__init__.py:24: in <module>
    from .framework import cilacc
../../../miniconda3/envs/mantidimaging-dev/lib/python3.12/site-packages/cil/framework/framework.py:43: in <module>
    cilacc = ctypes.cdll.LoadLibrary(dll)
../../../miniconda3/envs/mantidimaging-dev/lib/python3.12/ctypes/__init__.py:460: in LoadLibrary
    return self._dlltype(name)
../../../miniconda3/envs/mantidimaging-dev/lib/python3.12/ctypes/__init__.py:379: in __init__
    self._handle = _dlopen(self._name, mode)
E   OSError: libippcore.so.10: cannot open shared object file: No such file or directory
```

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -34,3 +34,4 @@ dependencies:
   - pydata-sphinx-theme==0.15.*
   - sphinx-copybutton==0.5.*
   - sphinx-design==0.6.*
+  - ipp=2021.12 # https://github.com/mantidproject/mantidimaging/issues/2354

--- a/environment.yml
+++ b/environment.yml
@@ -9,3 +9,4 @@ channels:
 dependencies:
   - mantidimaging>=2.5.0
   - conda-forge::numexpr # https://github.com/mantidproject/mantidimaging/issues/1774
+  - ipp=2021.12 # https://github.com/mantidproject/mantidimaging/issues/2354

